capturecat-core/src/main/java/com/capturecat/core/config/jwt/JwtFilter.java

```diff
@@ -0,0 +1,83 @@
+package com.capturecat.core.config.jwt;
+
+import java.io.IOException;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.security.SignatureException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import com.capturecat.core.config.auth.LoginUser;
+import com.capturecat.core.support.error.ErrorType;
+import com.capturecat.core.support.response.ApiResponse;
+
+@Slf4j
+@RequiredArgsConstructor
+public class JwtFilter extends OncePerRequestFilter {
+
+	private final JwtUtil jwtUtil;
+	private final ObjectMapper objectMapper = new ObjectMapper();
+
+	@Override
+	protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+		FilterChain filterChain) throws ServletException, IOException {
+
+		// Authorization 헤더에서 "Bearer <token>" 추출
+		String authHeader = request.getHeader("Authorization");
+
+		if (authHeader == null || !authHeader.startsWith("Bearer ")) {
+			log.info("Authorization header missing or malformed");
+			filterChain.doFilter(request, response); // 다음 필터로 넘김 (비인증 요청 허용할 수 있음)
+			return;
+		}
+
+		String accessToken = authHeader.substring(7); // "Bearer " 이후 토큰
+
+		//토큰 만료 검증. 만료 시 client에 즉시 응답. client는 재발급 요청 수행.
+		try {
+			jwtUtil.isExpired(accessToken);
+		} catch (ExpiredJwtException | SignatureException | MalformedJwtException | IllegalArgumentException e) {
+			rejectInvalidToken(response);
+			return;
+		}
+
+		// 토큰 유형 검사
+		if (!TokenType.ACCESS.name().equals(jwtUtil.getTokenType(accessToken))) {
```

Comment:
```suggestion
		if (JwtUtil.~) {
```

이 친구의 역할을 JwtUtil의 책임으로 넘기면 좋을 것 같아요. 그러면 조금 더 가독성이 좋은 메서드 명이 될 것 같아요! 예시 입니당.

```java
public boolean ~(String token) {
    return TokenType.ACCESS != getTokenType(toke);
}
```